resources/quicksight-subscriptions.go

```diff
@@ -0,0 +1,157 @@
+	"github.com/ekristen/libnuke/pkg/registry"
+	"github.com/ekristen/libnuke/pkg/resource"
+	"github.com/ekristen/libnuke/pkg/types"
+)
+
+const quickSightSubscriptionResource = "QuicksightSubscription"
+const subscriptionNameWhenNotAvailable = "NOT_AVAILABLE"
+
+func init() {
+	registry.Register(&registry.Registration{
+		Name:   quickSightSubscriptionResource,
+		Scope:  nuke.Account,
+		Lister: &QuickSightSubscriptionLister{},
+	})
+}
+
+type QuickSightSubscriptionLister struct {
+	stsService        stsiface.STSAPI
+	quicksightService quicksightiface.QuickSightAPI
+}
+
+type QuicksightSubscription struct {
+	svc               quicksightiface.QuickSightAPI
+	accountId         *string
+	name              *string
+	notificationEmail *string
+	edition           *string
+	status            *string
+}
+
+func (listener *QuickSightSubscriptionLister) List(_ context.Context, o interface{}) ([]resource.Resource, error) {
+	opts := o.(*nuke.ListerOpts)
+
+	var stsSvc stsiface.STSAPI
+	if listener.stsService != nil {
+		stsSvc = listener.stsService
+	} else {
+		stsSvc = sts.New(opts.Session)
+	}
+
+	var quicksightSvc quicksightiface.QuickSightAPI
+	if listener.quicksightService != nil {
+		quicksightSvc = listener.quicksightService
+	} else {
+		quicksightSvc = quicksight.New(opts.Session)
+	}
+
+	callerID, err := stsSvc.GetCallerIdentity(&sts.GetCallerIdentityInput{})
+	if err != nil {
+		return nil, err
+	}
+	accountId := callerID.Account
+
+	var resources []resource.Resource
+
+	describeSubscriptionOutput, err := quicksightSvc.DescribeAccountSubscription(&quicksight.DescribeAccountSubscriptionInput{
+		AwsAccountId: accountId,
+	})
+
+	if err != nil {
+		var resoureceNotFoundException *quicksight.ResourceNotFoundException
+		if !errors.As(err, &resoureceNotFoundException) {
+			return nil, err
+		}
+		return resources, nil
+	}
+
+	//The account name is only available some time later after the Subscription creation.
+	subscriptionName := subscriptionNameWhenNotAvailable
+	if describeSubscriptionOutput.AccountInfo.AccountName != nil {
+		subscriptionName = *describeSubscriptionOutput.AccountInfo.AccountName
+	}
+
+	resources = append(resources, &QuicksightSubscription{
+		svc:               quicksightSvc,
+		accountId:         accountId,
+		name:              &subscriptionName,
+		notificationEmail: describeSubscriptionOutput.AccountInfo.NotificationEmail,
+		edition:           describeSubscriptionOutput.AccountInfo.Edition,
+		status:            describeSubscriptionOutput.AccountInfo.AccountSubscriptionStatus,
+	})
+
+	return resources, nil
+}
+
+func (subscription *QuicksightSubscription) Remove(_ context.Context) error {
+	terminateProtectionEnabled := false
+
+	describeSettingsOutput, err := subscription.svc.DescribeAccountSettings(&quicksight.DescribeAccountSettingsInput{
+		AwsAccountId: subscription.accountId,
+	})
+	if err != nil {
+		return err
+	}
+
+	if *describeSettingsOutput.AccountSettings.TerminationProtectionEnabled {
```

Comment:
This one is a bit more work. I'd like to follow the pattern laid out by EC2Instance/RDSInstance and have a `Setting` on the resource called `DisableTerminationProtection`, then do a check for if it's `true` and if true, and it is enabled, then disable it before the remove.

All other resources that have some sort of termination protection use settings that must be configured.

Reply:
Hey @ekristen, thanks for the feedback. I'm using the DisableTerminationProtection setting now.